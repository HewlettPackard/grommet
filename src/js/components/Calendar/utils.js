
// Utility functions for the Calendar.
// Just what's needed to avoid having to include a dependency like momentjs.

const DAY_MILLISECONDS = 24 * 60 * 60 * 1000;

export const addDays = (date, days) =>
  (new Date(date.getTime() + (DAY_MILLISECONDS * days)));

export const subtractDays = (date, days) => addDays(date, -days);

export const addMonths = (date, months) => {
  const result = new Date(date);
  const years = Math.floor((date.getMonth() + months) / 12);
  result.setFullYear(date.getFullYear() + years);
  result.setMonth((date.getMonth() + months) % 12);
  return result;
};

export const subtractMonths = (date, months) => addMonths(date, -months);

export const sameDay = (date1, date2) => (
  date1.getFullYear() === date2.getFullYear() &&
  date1.getMonth() === date2.getMonth() &&
  date1.getDate() === date2.getDate()
);

export const sameDayOrAfter = (date1, date2) =>
  (date1.getFullYear() > date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() > date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() >= date2.getDate())
      )
    ))
  );

export const sameDayOrBefore = (date1, date2) =>
  (date1.getFullYear() < date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() < date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() <= date2.getDate())
      )
    ))
  );

export const daysApart = (date1, date2) =>
  Math.floor((date1.getTime() - date2.getTime()) / DAY_MILLISECONDS);

// betweenDates takes and array of two elements and checks if the
// supplied date lies between them, inclusive.
// returns 2 if exact match to one end, 1 if between, undefined otherwise
export const betweenDates = (date, dates) => {
  let result;
  if (dates) {
    const [from, to] = dates.map(d => new Date(d));
    if (sameDay(date, from) || sameDay(date, to)) {
      result = 2;
    } else if (sameDayOrAfter(date, from) && sameDayOrBefore(date, to)) {
      result = 1;
    }
  } else {
    result = 1;
  }
  return result;
};

// withinDates takes and array of string dates or 2 element arrays and
// checks whether the supplied date matches any string or is between
// any dates in arrays.
// returns 2 if exact match, 1 if between, undefined otherwise
export const withinDates = (date, dates) => {
  let result;
  if (dates) {
    if (Array.isArray(dates)) {
      dates.some((d) => {
        if (typeof d === 'string') {
          if (sameDay(date, new Date(d))) {
            result = 2;
          }
        } else {
          result = betweenDates(date, d);
        }
        return result;
      });
    } else if (sameDay(date, new Date(dates))) {
      result = 2;
    }
  }
  return result;
};
