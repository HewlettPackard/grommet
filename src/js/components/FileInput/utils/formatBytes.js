const SI_CONVERSION_FACTOR = 1000;
const IEC_CONVERSION_FACTOR = 1024;

const getBrowser = () => {
  const currentOS = ['Win', 'Linux', 'Mac'].find(
    (v) => window.navigator.userAgent.indexOf(v) >= 0,
  );

  return currentOS;
};

const defaultFormat = (size) => {
  const units = ['B', 'KB', 'MB', 'GB', 'TB'];
  const factor = SI_CONVERSION_FACTOR;
  let index = 0;
  let num = size;
  while (num >= factor && index < units.length - 1) {
    num /= factor;
    index += 1;
  }
  return `${num.toFixed(1)} ${units[index]}`;
};

const windowsFormat = (size) => {
  const num = Math.ceil(size / IEC_CONVERSION_FACTOR);

  function formatNumber(number) {
    return number.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1.');
  }

  return `${formatNumber(num)} ${'KB'}`;
};

const makeFormatBytes = (browser) => (size) => {
  switch (browser) {
    case 'Win':
      return windowsFormat(size);
    default:
      return defaultFormat(size);
  }
};

const formatBytes = makeFormatBytes(getBrowser());

export { formatBytes };
