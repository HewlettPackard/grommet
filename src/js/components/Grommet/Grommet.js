import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
