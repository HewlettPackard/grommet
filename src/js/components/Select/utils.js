import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { Box } from '../Box';
import { Button } from '../Button';
import { DropButton } from '../DropButton';
import { Text } from '../Text';
import { TextInput } from '../TextInput';
import {
  getHoverIndicatorStyle,
  selectedStyle,
  controlBorderStyle,
  normalizeColor,
} from '../../utils';

export const applyKey = (option, key) => {
  if (option === undefined) return undefined;
  if (typeof key === 'object') return applyKey(option, key.key);
  if (typeof key === 'function') return key(option);
  if (key !== undefined) return option[key];
  return option;
};

// position relative is so scroll can be managed correctly
export const OptionsBox = styled.div`
  position: relative;
  scroll-behavior: smooth;
  overflow: auto;
  outline: none;
`;

export const HiddenInput = styled.input`
  display: none;
`;

export const SelectOption = styled(Button)`
  ${(props) => props.selected && props.textComponent && selectedStyle}
  // applies theme.global.hover.background to the active
  // option for mouse and keyboard interactions
  ${(props) =>
    props.active &&
    getHoverIndicatorStyle(
      !props.children && !props.theme.select.options ? undefined : 'background',
      props.theme,
    )}
  display: block;
  width: 100%;
`;

export const SelectTextInput = styled(TextInput)`
  cursor: ${(props) => (props.defaultCursor ? 'default' : 'pointer')};
`;

export const StyledSelectDropButton = styled(DropButton)`
  ${(props) => !props.callerPlain && controlBorderStyle};
  ${(props) =>
    props.theme.select &&
    props.theme.select?.control &&
    props.theme.select?.control?.extend};
  ${(props) => props.open && props.theme.select?.control?.open};
`;

export const DefaultSelectTextInput = forwardRef(
  (
    { a11yTitle, disabled, id, name, placeholder, value, size, theme, ...rest },
    ref,
  ) => (
    <SelectTextInput
      a11yTitle={a11yTitle}
      // When Select is disabled, we want to show a default cursor
      // but not have disabled styling come from TextInput
      // Disabled can be a bool or an array of options to disable.
      // We only want to disable the TextInput if the control
      // button should be disabled which occurs when disabled
      // equals true.
      defaultCursor={disabled === true || undefined}
      focusIndicator={false}
      id={id ? `${id}__input` : undefined}
      name={name}
      ref={ref}
      {...rest}
      tabIndex="-1"
      type="text"
      placeholder={placeholder}
      plain
      readOnly
      value={value}
      size={size}
      theme={theme}
    />
  ),
);

export const EmptySearchOption = ({
  emptySearchMessage,
  selectOptionsStyle,
  theme,
}) => (
  <SelectOption
    key="search_empty"
    tabIndex="0"
    role="menuitem"
    hoverIndicator="background"
    disabled
    aria-live="polite"
  >
    <Box {...selectOptionsStyle}>
      <Text {...theme.select.container.text}>{emptySearchMessage}</Text>
    </Box>
  </SelectOption>
);

export const getOptionLabel = (index, options, labelKey) =>
  applyKey(options[index], labelKey);

export const getOptionValue = (index, options, valueKey) =>
  applyKey(options[index], valueKey);

export const checkDisabled = (
  index,
  disabled,
  disabledKey,
  options,
  valueKey,
) => {
  const option = options[index];
  let result;
  if (disabledKey) {
    result = applyKey(option, disabledKey);
  } else if (Array.isArray(disabled)) {
    if (typeof disabled[0] === 'number') {
      result = disabled.indexOf(index) !== -1;
    } else {
      const optionVal = getOptionValue(index, options, valueKey);
      result = disabled.indexOf(optionVal) !== -1;
    }
  }
  return result;
};

export const calcValuedValue = (value, valueKey) => {
  if (Array.isArray(value))
    return value.map((v) =>
      valueKey && valueKey.reduce ? v : applyKey(v, valueKey),
    );
  return valueKey && valueKey.reduce ? value : applyKey(value, valueKey);
};

export const changeEvent = (inputRef, nextValue) => {
  // Calling set value function directly on input because React library
  // overrides setter `event.target.value =` and loses original event
  // target fidelity.
  // https://stackoverflow.com/a/46012210
  const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
    window.HTMLInputElement.prototype,
    'value',
  ).set;
  nativeInputValueSetter.call(inputRef.current, nextValue);
  const event = new Event('input', { bubbles: true });
  inputRef.current.dispatchEvent(event);
};

export const getSelectIcon = (icon, theme, open) => {
  let SelectIcon;
  switch (icon) {
    case false:
      break;
    case true:
    case undefined:
      SelectIcon =
        open && theme.select.icons.up
          ? theme.select.icons.up
          : theme.select.icons.down;
      break;
    default:
      SelectIcon = icon;
  }
  return SelectIcon;
};

// if labelKey is a function and valueLabel is not defined
// we should use the labelKey function to display the
// selected value
export const getDisplayLabelKey = (
  labelKey,
  allOptions,
  optionIndexesInValue,
  selectValue,
) => {
  const optionLabelKey = applyKey(
    allOptions[optionIndexesInValue[0]],
    labelKey,
  );
  if (
    !selectValue &&
    optionIndexesInValue.length === 1 &&
    typeof optionLabelKey === 'object'
  )
    return optionLabelKey;
  return undefined;
};

export const getIconColor = (theme) =>
  normalizeColor(theme.select.icons.color || 'control', theme);
