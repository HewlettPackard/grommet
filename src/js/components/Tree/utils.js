export const setupNodes = nodes =>
  nodes.map(node => ({
    name: node.name,
    childs: setupNodes(node.childs || []),
    opened: false,
  }));

export const isSameNode = (node1, node2) => node1 === node2;

export const changeNode = (positions, nodes) => {
  const [currentPosition, ...otherPositions] = positions;
  const currentNode = nodes[currentPosition];

  if (otherPositions.length) {
    return nodes.map(node =>
      isSameNode(node, currentNode)
        ? {
            ...node,
            childs: changeNode(otherPositions, node.childs),
          }
        : { ...node },
    );
  }

  return nodes.map(node =>
    isSameNode(node, currentNode)
      ? {
          ...node,
          opened: !node.opened,
        }
      : { ...node },
  );
};
