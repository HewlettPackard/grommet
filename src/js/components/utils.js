import { css } from 'styled-components';

export const focusStyle = css`
  border-color: ${
    props => (
      props.theme.global.focus.border.color ||
      props.theme.global.colors.accent[0]
    )
  };
  box-shadow: 0 0 1px 1px ${
    props => (
      props.theme.global.focus.border.color ||
      props.theme.global.colors.accent[0]
    )
  };
`;

export function parseMetricToInt(fontAsString) {
  return parseInt(fontAsString.replace(/[^0-9]/g, ''), 10);
}

export function findAllByType(component, type) {
  let matches = [];

  if (component.type === type) {
    matches.push(component);
  }

  if (component.children) {
    component.children.forEach((child) => {
      matches = matches.concat(findAllByType(child, type));
    });
  }

  return matches;
}

export default { findAllByType, focusStyle, parseMetricToInt };
