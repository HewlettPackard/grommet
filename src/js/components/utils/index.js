export * from './styles';
export * from './mixins';
export * from './DOM';
export { default as Drop } from './Drop';
