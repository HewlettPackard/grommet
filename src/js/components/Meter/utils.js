import { colorForName } from '../../utils';

export const strokeProps = (color, theme) => {
  const result = {};
  if (color) {
    if (typeof color === 'object') {
      result.stroke = colorForName(color.color, theme);
      if (color.opacity) {
        result.strokeOpacity = (color.opacity === true ?
          theme.global.opacity.medium : theme.global.opacity[color.opacity]);
      }
    } else {
      result.stroke = colorForName(color, theme);
    }
  }
  return result;
};

export default { strokeProps };
