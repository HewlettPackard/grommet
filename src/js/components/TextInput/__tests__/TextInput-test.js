import React from 'react';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { createPortal, expectPortal } from '../../../utils/portal';

import { Grommet } from '../../Grommet';
import { TextInput } from '../';

Enzyme.configure({ adapter: new Adapter() });

describe('TextInput', () => {
  beforeEach(createPortal);

  test('mounts', () => {
    const component = mount(
      <TextInput name='item' />
    );
    expect(component.getDOMNode()).toMatchSnapshot();
  });

  test('mounts with suggestions', () => {
    const onInput = jest.fn();
    const onFocus = jest.fn();
    const component = mount(
      <TextInput
        id='item'
        name='item'
        suggestions={['test', 'test1']}
        onInput={onInput}
        onFocus={onFocus}
      />
    );
    const input = component.find('input').first();
    input.simulate('input');
    input.simulate('focus');
    expectPortal('text-input-drop__item').toMatchSnapshot();
    expect(onInput).toBeCalled();
    expect(onFocus).toBeCalled();

    global.document.dispatchEvent(new Event('click'));

    expect(document.getElementById('text-input-drop__item')).toBeNull();
  });

  test('mounts with complex suggestions', () => {
    const component = mount(
      <Grommet>
        <TextInput
          id='item'
          name='item'
          suggestions={[
            { label: 'test', value: 'test' },
            { value: 'test1' },
          ]}
        />
      </Grommet>
    );
    component.find('input').first().simulate('input');
    expectPortal('text-input-drop__item').toMatchSnapshot();
  });

  test('closes suggestion drop', () => {
    const component = mount(
      <Grommet>
        <TextInput
          id='item'
          name='item'
          suggestions={['test', 'test1']}
        />
      </Grommet>
    );

    const input = component.find('input').first();
    input.simulate('input');

    expectPortal('text-input-drop__item').toMatchSnapshot();

    input.simulate('keyDown', { key: 'Esc', keyCode: 27, which: 27 });

    expect(document.getElementById('text-input-drop__item')).toBeNull();
    expect(component.getDOMNode()).toMatchSnapshot();
  });

  test('selects suggestion', () => {
    const onSelect = jest.fn();
    const component = mount(
      <Grommet>
        <TextInput
          plain={true}
          size='large'
          id='item'
          name='item'
          suggestions={['test', 'test1']}
          onSelect={onSelect}
        />
      </Grommet>
    );

    const input = component.find('input').first();
    input.simulate('input');

    expectPortal('text-input-drop__item').toMatchSnapshot();

    document.getElementById('text-input-drop__item').querySelector('button').click();

    expect(component.getDOMNode()).toMatchSnapshot();
    expect(document.getElementById('text-input-drop__item')).toBeNull();
    expect(onSelect).toBeCalled();
  });

  test('handles next and previous suggestions', () => {
    const component = mount(
      <Grommet>
        <TextInput
          id='item'
          name='item'
          suggestions={['test', { value: 'test1' }]}
        />
      </Grommet>
    );

    const input = component.find('input').first();
    const preventDefault = jest.fn();
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 38,
      preventDefault,
    });
    expect(preventDefault).toBeCalled();
  });

  test('selects a suggestion', () => {
    const onSelect = jest.fn();
    const component = mount(
      <Grommet>
        <TextInput
          id='item'
          name='item'
          suggestions={['test', { value: 'test1' }]}
          onSelect={onSelect}
        />
      </Grommet>
    );

    const input = component.find('input').first();
    const preventDefault = jest.fn();
    // pressing enter here nothing will happen
    input.simulate('keyDown', {
      keyCode: 13,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 13,
      preventDefault,
    });
    expect(preventDefault).toBeCalled();
    expect(onSelect).toBeCalled();
  });

  test('handles next and previous without suggestion', () => {
    const component = mount(
      <Grommet>
        <TextInput
          id='item'
          name='item'
        />
      </Grommet>
    );
    const input = component.find('input').first();
    const preventDefault = jest.fn();
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 40,
      preventDefault,
    });
    input.simulate('keyDown', {
      keyCode: 38,
      preventDefault,
    });
    expect(preventDefault).not.toBeCalled();
    expect(component.getDOMNode()).toMatchSnapshot();
  });
});
