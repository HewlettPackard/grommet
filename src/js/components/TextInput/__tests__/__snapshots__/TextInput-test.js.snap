// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput closes suggestion drop 1`] = `
<div
  class="StyledDrop-bCWxPe gcEofR"
  tabindex="-1"
  id="text-input-drop__item"
  style="left: 0px; width: 0.1px; top: 0px; max-height: 768px;"
>
  <ol
    class="StyledTextInput__StyledSuggestions-bNrBob bCeRym"
  >
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test
        </div>
      </button>
    </li>
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test1
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`TextInput closes suggestion drop 2`] = `
"@media only screen and (min-width:700px) {
  .dYRvzO {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}



.gcEofR {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  border-radius: 0px;
  box-shadow: 0px 3px 8px rgba(100,100,100,0.50);
  position: fixed;
  z-index: 20;
  overflow: auto;
  outline: none;
  background-color: #f8f8f8;
  color: #333333;
  opacity: 0;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-animation: jNdvGQ 0.1s forwards;
  animation: jNdvGQ 0.1s forwards;
  -webkit-animation-delay: 0.01s;
  animation-delay: 0.01s;
}"
`;

exports[`TextInput closes suggestion drop 3`] = `
<div
  class="StyledGrommet-gGRXwz deqpVW"
>
  <div
    class="StyledTextInput__StyledTextInputContainer-bPTWQl hasTun"
  >
    <input
      class="StyledTextInput-bzOzsW zdvFG"
      id="item"
      autocomplete="off"
      name="item"
    />
  </div>
</div>
`;

exports[`TextInput handles next and previous without suggestion 1`] = `
<div
  class="StyledGrommet-gGRXwz deqpVW"
>
  <div
    class="StyledTextInput__StyledTextInputContainer-bPTWQl hasTun"
  >
    <input
      class="StyledTextInput-bzOzsW zdvFG"
      id="item"
      autocomplete="off"
      name="item"
    />
  </div>
</div>
`;

exports[`TextInput mounts 1`] = `
<div
  class="StyledTextInput__StyledTextInputContainer-bPTWQl hasTun"
>
  <input
    class="StyledTextInput-bzOzsW zdvFG"
    autocomplete="off"
    name="item"
  />
</div>
`;

exports[`TextInput mounts with complex suggestions 1`] = `
<div
  class="StyledDrop-bCWxPe gcEofR"
  tabindex="-1"
  id="text-input-drop__item"
  style="left: 0px; width: 0.1px; top: 0px; max-height: 768px;"
>
  <ol
    class="StyledTextInput__StyledSuggestions-bNrBob bCeRym"
  >
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test
        </div>
      </button>
    </li>
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test1
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`TextInput mounts with complex suggestions 2`] = `
"@media only screen and (min-width:700px) {
  .dYRvzO {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}



.gcEofR {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  border-radius: 0px;
  box-shadow: 0px 3px 8px rgba(100,100,100,0.50);
  position: fixed;
  z-index: 20;
  overflow: auto;
  outline: none;
  background-color: #f8f8f8;
  color: #333333;
  opacity: 0;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-animation: jNdvGQ 0.1s forwards;
  animation: jNdvGQ 0.1s forwards;
  -webkit-animation-delay: 0.01s;
  animation-delay: 0.01s;
}"
`;

exports[`TextInput mounts with suggestions 1`] = `
<div
  class="StyledDrop-bCWxPe gcEofR"
  tabindex="-1"
  id="text-input-drop__item"
  style="left: 0px; width: 0.1px; top: 0px; max-height: 768px;"
>
  <ol
    class="StyledTextInput__StyledSuggestions-bNrBob bCeRym"
  >
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test
        </div>
      </button>
    </li>
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test1
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`TextInput mounts with suggestions 2`] = `
"@media only screen and (min-width:700px) {
  .dYRvzO {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}



.gcEofR {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  border-radius: 0px;
  box-shadow: 0px 3px 8px rgba(100,100,100,0.50);
  position: fixed;
  z-index: 20;
  overflow: auto;
  outline: none;
  background-color: #f8f8f8;
  color: #333333;
  opacity: 0;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-animation: jNdvGQ 0.1s forwards;
  animation: jNdvGQ 0.1s forwards;
  -webkit-animation-delay: 0.01s;
  animation-delay: 0.01s;
}"
`;

exports[`TextInput selects suggestion 1`] = `
<div
  class="StyledDrop-bCWxPe gcEofR"
  tabindex="-1"
  id="text-input-drop__item"
  style="left: 0px; width: 0.1px; top: 0px; max-height: 768px;"
>
  <ol
    class="StyledTextInput__StyledSuggestions-bNrBob bCeRym"
  >
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test
        </div>
      </button>
    </li>
    <li>
      <button
        class="StyledButton-gXzblK dYRvzO"
        type="button"
      >
        <div
          class="StyledBox-bStriS gedxom"
          direction="column"
        >
          test1
        </div>
      </button>
    </li>
  </ol>
</div>
`;

exports[`TextInput selects suggestion 2`] = `
"@media only screen and (min-width:700px) {
  .dYRvzO {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}



.gcEofR {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  border-radius: 0px;
  box-shadow: 0px 3px 8px rgba(100,100,100,0.50);
  position: fixed;
  z-index: 20;
  overflow: auto;
  outline: none;
  background-color: #f8f8f8;
  color: #333333;
  opacity: 0;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-animation: jNdvGQ 0.1s forwards;
  animation: jNdvGQ 0.1s forwards;
  -webkit-animation-delay: 0.01s;
  animation-delay: 0.01s;
}"
`;

exports[`TextInput selects suggestion 3`] = `
<div
  class="StyledGrommet-gGRXwz deqpVW"
>
  <div
    class="StyledTextInput__StyledTextInputContainer-bPTWQl hasTun"
  >
    <input
      class="StyledTextInput-bzOzsW gAbvjB"
      id="item"
      autocomplete="off"
      name="item"
    />
  </div>
</div>
`;
